controllers/machineset_controller.go

```diff
@@ -631,11 +631,25 @@ func (r *MachineSetReconciler) calculateStatus(ctx context.Context, cluster *clu
 		}
 	}
 
-	newStatus.Replicas = int32(len(filteredMachines))
-	newStatus.FullyLabeledReplicas = int32(fullyLabeledReplicasCount)
-	newStatus.ReadyReplicas = int32(readyReplicasCount)
-	newStatus.AvailableReplicas = int32(availableReplicasCount)
-	return newStatus, nil
+	newMS.Status.Replicas = int32(len(filteredMachines))
+	newMS.Status.FullyLabeledReplicas = int32(fullyLabeledReplicasCount)
+	newMS.Status.ReadyReplicas = int32(readyReplicasCount)
+	newMS.Status.AvailableReplicas = int32(availableReplicasCount)
+	replicaConditionPresent := conditions.Get(newMS, clusterv1.ReplicasUpdated) != nil
+	if replicaCountsDifferent(&newMS.Status, &ms.Status) || !replicaConditionPresent {
+		// Delete and set the condition to force-update lastTransitionTime
+		replicasUpdatedCondition := clusterv1.CreateReplicasUpdatedCondition()
+		if replicaConditionPresent {
+			conditions.Delete(newMS, clusterv1.ReplicasUpdated)
+		}
+		conditions.Set(newMS, &replicasUpdatedCondition)
```

Comment:
Instead of deleting/recreating the condition to force the update, would it be possible to just encode the replica count into the message of the condition? I would expect that to be enough to trigger an update to lastTransitionTime.

Reply:
Yeah, this is an interesting suggestion.  It would need to look like 'ReplicaCounts: A/B/C/D' or similar.  Each different amount needs to be recorded because any of them might change.  My only concern is, will this look confusing to the user?  I think this does solve needing to create a new method in conditions.Setter though.